### PR TITLE
MM-56540 - Type changes for live-captions-related refactoring

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- use `CallJobState` instead of `CallRecordingState`
- put the job type into `CallJobStateData`

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-56540

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -208,7 +209,8 @@ export type CallState = {
     screen_sharing_session_id: string;
     owner_id: string;
     host_id: string;
-    recording?: CallRecordingState;
+    recording?: CallJobState;
+    live_captions?: CallJobState;
```
Comment (streamer45) [inline src/types/types.ts:213]: So we are using nouns here :p

**Review by streamer45 — APPROVED** [review]

File: src/types/types.ts (hunk 1 above)
Comment (cpoile) [reply src/types/types.ts:213]: @streamer45 Hmm... Yah, looks like it -- I followed what we were doing https://github.com/mattermost/mattermost-plugin-calls/blob/4920d771d4783d084ae4b07ed171f2536b9c2eeb/server/state.go#L85-L86
`recording`, `transcription`, `live_captions`.
Up to you if you want to change it, now would be the time.

Comment (streamer45) [reply src/types/types.ts:213]: It's fine, I surrender, we can't really win entropy.

Comment (cpoile) [reply src/types/types.ts:213]: 😂

**Pull request merged**